serverenvironmentexample.adoc

```diff
@@ -0,0 +1,155 @@
+:icons:
+
+A GRR Server Environment can be whatever you make it. This example goes over the
+steps necessary to create a three server environment: one to house the front end HTTP
+service, one to house the master datastore, and one to house the admin ui and
+workers.
+
+Assumptions Made
+----------------
+Below are the assumptions made to make use of this example.
+
+* GRR Servers will be running Ubuntu 16.04 LTS with all updates already installed
+* There will be 3 servers in the configuration as outlined above. They will be called Master Dataserver, HTTP Server, and Main Server
+* Bleeding edge GRR is desired and will be installed via pip
+* The virtual environment will be called GRR_NEW
+* All networking has been properly set up
+
+Setting up the servers
+----------------------
+The steps below are based off of
+https://github.com/google/grr-doc/blob/master/installfrompip.adoc[Install from
+Pip] documentation. If the below does not work, please refer back to there.
+
+1. `apt-get install debhelper dpkg-dev libssl-dev python-dev python-pip rpm git
+protobuf-compiler`
+2. `pip install --upgrade pip`
+3. `pip install virtualenv`
+4. `git clone https://github.com/google/grr.git`
+5. `virtualenv GRR_NEW`
+6. `source GRR_NEW/bin/activate`
+7. `cd grr/`
+8. `pip install --editable .`
+9. `pip install --editable grr/config/grr-response-server`
+10. `pip install --no-cache-dir -f
+https://storage.googleapis.com/releases.grr-response.com/index.html
+grr-response-templates`
+11. `pip install --editable grr/config/grr-response-test`
+
+Configuring GRR on the Main Server
+----------------------------------
+1. Set up a MTA of your choice
+2. Run `grr_config_updater initialize`
+* Step 1: Key Generation: - this step autocompletes and needs no user input
+* Step 2: Setting Basic Configuration Parameters:
+    - GRR Datastore: Choose `1` for a SQLite Datastore. That is the datastore
+      all these tutorials assume you have
+    - GRR URLs: Enter hostname: `<main server ip>`
+    - Server URL: Frontend URL `http://<http server ip>:8080/`
+    - AdminUI URL: leave as default, press enter
+    - GRR Emails: Email Domain: `<your email domain>`
+    - GRR Emails: Alert Email Address: `<your alerts email>`
+    - GRR Emails: Emergency Email Access Email Address: `<your emergency email>`
+* Step 3: Adding Admin User: `<password>`
+* Step 4: Installing template package: `n`
+* Step 5: - Completes on its own
+3. _Optional_ - Edit `grr/install_data/etc/server.local.yaml` and move
+`PrivateKeys.executable_signing_private_key` key to a new file and change permissions
+based on your executable signing structure policy
+* When you move the key to a new file, remove all extra whitespace as well.
+4. _Optional_ - Set up the Approval Process
+* Users listed in the approval process will be added later. However this configuration must be pushed out to all servers, so it should be set up now.
+* https://github.com/google/grr-doc/blob/master/admin.adoc#auditing[Auditing]
+
```

Comment:
Thanks for writing this up. Everything up to this point is mostly a duplicate of existing docs (pip) or the install script prompts. I don't think we want to add that duplication since one part will inevitably be out of date - it's hard enough maintaining one place. I'm already thinking about modifying the prompts for the next release and don't want to have to update it here as well.